web/pdf_viewer.js

```diff
@@ -1851,6 +1862,18 @@ class PDFViewer {
     );
 
     if (pageView) {
+      if (
+        this.#scrollTimeoutId !== null &&
+        pageView instanceof PDFPageDetailView &&
+        pageView.renderingCancelled
+      ) {
+        // If we are scrolling and the rendering of the detail view was just
+        // cancelled, it's because the user is scrolling too quickly and so
+        // we constantly need to re-render a different area.
+        // Don't attempt to re-rendering it: this will be done once the user
+        // stops scrolling.
+        return true;
+      }
```

Comment:
I don't know how much it matters in practice, but won't returning here without trigger rendering effectively stop any possible pre-rendering of other *currently invisible* pages?

Until either another page becomes the more visible *or* scrolling stops, it seems that the `getHighestPriority`-method will [now always break here](https://github.com/mozilla/pdf.js/pull/19128/files#diff-06d001972b1007eb10259e1678f748a90ad984af2ab252a1125d4387d696f61cR132-R134).

Would it make (any) sense to e.g. set the detailView-renderingState to `FINISHED` when `renderingCancelled === true` such that other pre-rendering can still run?

Reply:
I fixed this in a slightly different way, as when marking the view as finished then we don't re-render it unless the new area is enough different from the old one.

Instead, `getHighestPriority` now receives a parameter to make it ignore detail views.